### ISSUE TITLE
How to use the configuration script to register multiple hosts for multiple services?

### ISSUE THREAD

**Issue by InteXX** [ohadschn/letsencrypt-webapp-renewer#107]
I'm looking through the source code for the [configuration script](https://github.com/ohadschn/letsencrypt-webapp-renewer/blob/master/src/Scripts/Set-LetsEncryptConfiguration.ps1) and I'm a bit cornfused about something.

What syntax should we use for multiple hostnames per app service when we're specifying multiple services for the `WebApp` parameter?

For example, if we have two services `foo` and `bar`, and we want a multiple-host SAN cert for each one, like so:

```
foo.com
www.foo.com

bar.com
www.bar.com
```

...how do we indicate this in the script parameters?

Should we do something like this:

```
.\Set-LetsEncryptConfiguration.ps1 -WebApp "foo;bar" -Hosts "foo.com;www.foo.com;bar.com;www.bar.com"
```

It's unclear to me how to proceed.

**Comment by InteXX** [comment]
I'm going to update the script for this; I'll send over a PR when it's ready.

**Comment by InteXX** [comment]
@ohadschn 

My proposed update will assume this construct:

```
letsencrypt:webApps = foo;bar
letsencrypt:foo-hosts = foo.com;www.foo.com
letsencrypt:bar-hosts = bar.com;www.bar.com
```

Will this parse correctly?

**Comment by ohadschn** [comment]
Yeah the script was built with a single web app / slot / group in mind, you need multiple groups - a PR would be great thanks!

The construct you pasted looks good, when in doubt you can check the sample config: https://github.com/ohadschn/letsencrypt-webapp-renewer#sample-configuration

PS in order to get the behavior from that other issue (cert per host and as a side effect continue on error at the host level), you would need to use groups: https://github.com/ohadschn/letsencrypt-webapp-renewer#multiple-certificates-for-a-single-site. Personally I don't bother with such overhead, but up to you.

**Comment by InteXX** [comment]
I'm over my head with overhead, so I don't suppose I'll bother either ;-)

How would a person go about achieving this same construct using the CLI? Multiple `-o` switches perhaps?

**Comment by InteXX** [comment]
I have new certs.

Thank you, Ohad, you've saved me :-)

**Comment by ohadschn** [comment]
Great news :)
A couple of clarifications:
1. In case it wasn't clear, the PowerShell script is only for convenience (and possible configuration modification automation). You can just set all the configuration you need manually in the portal (that's what I do) - see screenshot below.
1. I added a sample for CLI execution to the README. As you have probably discovered by now, multiple `-o` switches should work (though personally I prefer passing a semi-colon delimited list for consistency with Web App settings).
1. Revisiting the script, it looks like you can just execute it multiple times for multiple WebApps: https://github.com/ohadschn/letsencrypt-webapp-renewer/blob/7412988b82e556858028186a8b233fb1cad4e90c/src/Scripts/Set-LetsEncryptConfiguration.ps1#L147

I see that the logic isn't perfect though (simple substring search that will fail in cases like having webapps `foo` and `foo-bar`) so I'll keep this issue open. We can also consider native script support for multiple web apps, possibly via a second script that would run the first.

![image](https://user-images.githubusercontent.com/8555641/88394760-a1cde500-cdc8-11ea-8be4-df9812465dca.png)

**Comment by InteXX** [comment]
OK, sounds good.

I'm having trouble getting this PR to you. My proposed script updates are ready, but clicking the New Pull Request link on a new branch in VS 2019 produces no result. Further, attempting to create it in my browser via this repo results in an error when I upload the file:

> We don’t support that file type. Try again with a GIF, JPEG, JPG, PNG, DOCX, GZ, LOG, PDF, PPTX, TXT, XLSX or ZIP.

This is my first pull request, anywhere, ever. You saw it happen here :-)

Any ideas?

**Comment by InteXX** [comment]
> I'm having trouble getting this PR to you

OK, got it.

RTFM ;-)
